src/main/java/com/rybka/controller/ExchangePageController.java

```diff
@@ -0,0 +1,54 @@
+package com.rybka.controller;
+
+import com.rybka.constant.EndpointConstants;
+import com.rybka.model.CurrencyHistory;
+import com.rybka.model.dto.ExchangeRequest;
+import com.rybka.repository.CurrencyHistoryRepository;
+import com.rybka.service.exchange.ExchangeService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+@RequestMapping(value = {EndpointConstants.ROOT_URL, EndpointConstants.EXCHANGE_ROOT_URL})
+@RequiredArgsConstructor
+public class ExchangePageController {
+    private final ExchangeService service;
+    private final CurrencyHistoryRepository repository;
+
+    @GetMapping
+    public String showExchangePage(Model model) {
+        model.addAttribute("exchangeRequest", new ExchangeRequest());
+
+        return "exchangePage";
+    }
+
+    @PostMapping(EndpointConstants.EXCHANGE_POST_URL)
```

Comment:
/banks/result 

/banks/rates - will be better/

Reply:
fixed